component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandler.java

```diff
@@ -45,128 +41,89 @@
 public class BackupCodeAPIHandler {
 
     /**
-     * Retrieve backup codes for the given username.
+     * Returns the number of backup codes remaining for the given user.
      *
      * @param username Username of the user.
-     * @return List of backup code of the user.
+     * @return the number of backup codes remaining for the given user.
      * @throws BackupCodeException If an error occurred while getting backup codes.
      */
-    public static List<String> getBackupCodes(String username) throws BackupCodeException {
+    public static int getRemainingBackupCodesCount(String username) throws BackupCodeException {
 
-        String tenantAwareUsername;
+        List<String> remainingBackupCodesList = new ArrayList<>();
         try {
             UserRealm userRealm = BackupCodeUtil.getUserRealm(username);
             if (userRealm != null) {
-                tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+                String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
                 Map<String, String> userClaimValues = userRealm.getUserStoreManager()
                         .getUserClaimValues(tenantAwareUsername, new String[]{BACKUP_CODES_CLAIM}, null);
-                String encryptedBackupCodes = userClaimValues.get(BACKUP_CODES_CLAIM);
-                String backupCodes = null;
-                if (StringUtils.isNotEmpty(encryptedBackupCodes)) {
-                    backupCodes = BackupCodeUtil.decrypt(encryptedBackupCodes);
-                }
-                if (StringUtils.isEmpty(backupCodes)) {
-                    return Collections.emptyList();
-                } else {
-                    return new ArrayList<>(new ArrayList<>(Arrays.asList(backupCodes.split(","))));
+                String backupCodes = userClaimValues.get(BACKUP_CODES_CLAIM);
+                if (backupCodes != null && StringUtils.isNotBlank(backupCodes)) {
```

Comment:
Redundant check. `backupCodes != null` check is not needed.

Reply:
addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/fadf64b3e70925c84689808a42366845bae0660a](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/fadf64b3e70925c84689808a42366845bae0660a)